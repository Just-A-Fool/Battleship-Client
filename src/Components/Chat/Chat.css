.chat-username {
    margin:20px;
    margin-right: 10px;
    font-family: 'Roboto Condensed', sans-serif;
}

.chat-me {
    color: rgb(57, 206, 252);
    text-shadow: 0 10px 50px 0 rgba(9, 58, 138, 0.5);
}

.chat-opponent {
    color: red;
}

.chat-message {
    margin-bottom: 20px;
    margin-top: 20px;
    word-break: break-all;
    font-family: 'Roboto Condensed', sans-serif;
    
}

.chat-li {
    display: flex;
    flex-direction: row;
    margin: 0;
}

#chat-window {
    padding: 20px;
    border: 0.1rem solid black;
    margin: 0 auto;
    overflow: auto;
    height: 350px;
    background-color: rgb(0, 0, 0);
}

#chat {
    width: 90%;
    margin: 0 auto;
    padding: 5px;
    background-color: rgb(4, 12, 26);
}

#chatInput {
    text-align: center;
    border: 0.1rem solid rgb(59, 59, 59);
    background-color: black;
    color: white;
    width: 70%;
    height: 35px;
    font-family: 'Roboto Condensed', sans-serif;
}

#chat-label {
    color: white;
    font-family: 'Roboto Condensed', sans-serif;
}

#chat-form > fieldset {
    border: none;
}

@media screen and (min-width: 1025px) {
    #chat {
        width: 60%;
    }

    .chat-username,
    .chat-message,
    .chat-opponent,
    .chat-me {
        font-size: 1.5rem;
    }

    #chatInput {
        height: 40px;
        
    }

    #chat-window {
        height: 430px;
    }
}

@media screen and (min-width: 1300px) {
    #chat {
        width: 23%;
        position: absolute;
        top: 235px;
        z-index: 1;
    }

    li.chat-li {
        padding:0px;

    }

    .chat-message {
        margin-bottom: 5px;
        margin-top: 5px;
    }

    .chat-username {
        margin: 5px;
    }

    #chat-form > fieldset {
        border: none;
        font-size: 1.2rem;
    }

    #chat-form > fieldset > input {
        font-size: 1rem;
    }

    .chat-username,
    .chat-message,
    .chat-opponent,
    .chat-me {
        font-size: 1rem;
    }
        
    #chat-window {
    height: 400px;
    padding: 5px;
    }
    .outer {
        position: relative;
        overflow: hidden;
    }
    .grid-box.one-only{
        position: relative;
        left: 0px;
    }
    .grid-box{
        position: relative;
        left: 140px;
        z-index: 1; 
    }
}

@media screen and (min-width: 1750px) {
    #chat {
        width: 30%;
    }

    .grid-box {
        left: 200px;
    }
}
