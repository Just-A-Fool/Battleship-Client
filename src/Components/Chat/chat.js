import React from 'react';
import './Chat.css'


export default class Chat extends React.Component {
    state = {
        messages: [{username:'schmittywormenjagermanjensen', message:'hello'}, {username:'Me', message:'Hi yourself schmittywormenjagermanjensenschmittywormenjagermanjensenschmittywormenjagermanjensenschmittywormenjagermanjensenschmittywormenjagermanjensenschmittywormenjagermanjensenschmittywormenjagermanjensenschmittywormenjagermanjensenschmittywormenjagermanjensenschmittywormenjagermanjensen'}, {username:'schmittywormenjagermanjensen', message:'glhf'}]
    }

    componentDidMount = () => {
        this.props.socket.on('chat-message', data => {

            this.setState({
                messages: [...this.state.messages, {username: data.username, message: data.message}]
            })
            
        })
    }

    handleChatMessage = (message) => {
        
        this.props.socket.emit('send-message', {room: this.props.room, message: message});
        this.setState({
            messages: [...this.state.messages, {username: 'Me', message}]
        })
    }


    render() {

        let chatHistory = this.state.messages.map((message, index) => {
            return(
                <li className='chat-li' key={index}>
                    <h4 className={message.username === 'Me' ? 'chat-username chat-me' : 'chat-username chat-opponent'}>{message.username}</h4>
                    <p className='chat-message'>{message.message}</p>
                </li>
            )
        })




        return (
            <div id='chat'>
                <ul id='chat-window'>
                    {chatHistory}
                </ul>
                <form id='chat-form' onSubmit={(event) => {
                    event.preventDefault();
                    this.handleChatMessage(event.target.chatInput.value);
                    event.target.chatInput.value = '';
                }}>
                    <label htmlFor='chatInput'>Chat Input</label>
                    <input type='text' placeholder='Type your message here.' id='chatInput' />
                </form>
            </div>

        );
    }
}






// socketmessage in state

// clearSocketMessage=()=>{
//     this.setState({
//       socketMessage:''
//     })
//   }

//   socketMessage=(message)=>{
//     this.setState({
//       socketMessage:message
//     })
//   }

// socket.on('left', data =>{
//     this.setState({
//       socketMessage:data
//     })
//     setTimeout(()=>{
//       this.clearSocketMessage()}
//     , 3000)
//   })
//   roomName = this.state.room
//   console.log(roomName)