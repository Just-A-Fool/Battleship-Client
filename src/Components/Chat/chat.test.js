import React from 'react';
import ReactDOM from 'react-dom';
import Chat from './Chat';
import { shallow } from 'enzyme'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'

describe(`Chat component`, () => {

    it(`Renders without crashing`, () => {
        const div = document.createElement('div');
        ReactDOM.render(<Chat />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it(`Displays a chat window when rendered`, () => {
        const wrapper = shallow(<Chat />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it(`Expects an input where the user can type a message, with a placeholder identifying it as such` , () => {
        const wrapper = shallow(<Chat />)
        expect(wrapper.contains(<input type='text' placeholder='Type Here' id='chatInput' aria-label='type message to opponent here' />)).toEqual(true)
    })

    it(`Expects an a message window where the user can see messages from the other player` , () => {
        const wrapper = shallow(<Chat />)
        expect(wrapper.contains(<ul id='chat-window'></ul>)).toEqual(true)
    })

    it(`Expects a label for the chat input area`, () => { 
        const wrapper = shallow(<Chat />)
        expect(wrapper.text()).toMatch(/Chat with Opponent/)
    })

    it(`Stores messages received from the other player from data found in component state`, () => {
        const wrapper = shallow(<Chat />)
        let newMessage = 'Hello!'
        let temp = []
        temp.push(newMessage)
        wrapper.setState({messages:temp})
        expect(wrapper.state().messages[0]).toHaveLength(6)
    })

});


