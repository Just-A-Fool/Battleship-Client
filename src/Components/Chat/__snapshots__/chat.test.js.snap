// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat component Displays a chat window when rendered 1`] = `
<div
  aria-label="chat-box"
  aria-live="polite"
  id="chat"
>
  <ul
    id="chat-window"
  />
  <form
    id="chat-form"
    onSubmit={[Function]}
  >
     
    <fieldset>
      <legend>
        Chat with Opponent
      </legend>
      <label
        htmlFor="chatInput"
        id="chat-label"
      >
        Type Message
      </label>
      <input
        aria-label="type message to opponent here"
        id="chatInput"
        placeholder="Type Here"
        type="text"
      />
    </fieldset>
  </form>
</div>
`;
